test/test_broker.hpp

```diff
@@ -1011,7 +1046,7 @@ class test_broker {
             auto& idx = subs_.get<tag_con>();
             auto const& range = boost::make_iterator_range(idx.equal_range(spep));
             // In v3_1_1, session_expiry_interval is not set. So clean on close.
-            if (ep.clean_session() && session_clear) {
+            if (session_clear) {
```

Comment:
Won't this cause problems for 3_1_1 ?

Reply:
I think that it is no problem.
Because 3_1_1 is considered at
https://github.com/redboltz/mqtt_cpp/pull/657/files#diff-45ca9626e6c7c611be2ceb2ff63031a5R976
and reflected to the value session_clear.

Reply:
Ok